src/views/SetUnitView/SetUnitMounts/SetUnitMounts.tsx

```diff
@@ -23,50 +31,88 @@ const SetUnitMounts: FunctionComponent<Props> = ({
 
   return (
     <Grid
-      container
+      padding="8px 16px"
       marginBottom="16px"
       alignItems="center"
       justifyContent="space-between"
-      flexWrap="nowrap"
-      padding="8px 16px"
-      direction="row"
+      component={TransitionGroup}
       sx={{
-        width: "100%",
-        gap: "auto",
-        minHeight: theme.spacing(9),
-        borderRadius: "4px",
         background:
           "linear-gradient(180deg, rgba(255, 255, 255, 0.11) 0%, rgba(255, 255, 255, 0.11) 100%), #121212",
+        borderRadius: "4px"
       }}
     >
-      <Grid container direction="row">
-        <Grid
-          container
-          direction="row"
-          justifyContent="space-between"
-          alignItems="center"
-        >
+      <Grid
+        container
+        flexWrap="nowrap"
+        direction="row"
+        alignItems="center"
+        sx={{
+          width: "100%",
+          gap: "auto"
+        }}
+      >
+        <Grid container direction="row">
           <Grid
             container
-            direction="column"
-            justifyContent="flex-start"
-            wrap="wrap"
-            width="176px"
+            direction="row"
+            justifyContent="space-between"
+            alignItems="center"
           >
-            <Typography>{mounts}</Typography>
-            <Typography variant="body1" sx={{ color: "text.disabled" }}>
-              {data.mounts[mounts].Cost} Points
-            </Typography>
+            <Grid
+              container
+              direction="column"
+              justifyContent="flex-start"
+              wrap="wrap"
+              width="176px"
+            >
+              {data.mounts[mounts].Type.map((type, key) => (
+                <>
+                  <Typography
+                    variant="body2"
+                    key={key}
+                    sx={{ color: "text.disabled" }}
+                  >
+                    {type}
+                  </Typography>
+                </>
+              ))}
+              <Typography>{mounts.replace(/_/g, " ")}</Typography>
+              <Typography variant="body1" sx={{ color: "text.disabled" }}>
+                {data.mounts[mounts].Cost} Points
+              </Typography>
+            </Grid>
+            <SetUnitMountsStats mounts={mounts} />
           </Grid>
-          <SetUnitMountsStats mounts={mounts} />
         </Grid>
+
+        <Radio
+          checked={currentMounts === mounts}
+          value={mounts}
+          onClick={handleClick}
+          sx={{ height: 1 }}
+        />
       </Grid>
 
-      <Radio
-        checked={currentMounts === mounts}
-        value={mounts}
-        onClick={handleClick}
-      />
+      {currentMounts === mounts && (
+        <Grid component={Collapse}>
+          {data.mounts[mounts].Armour && (
+            <SetUnitMountArmour
+              armoury={data.mounts[mounts].Armour}
```

Comment:
Somethings up with the types here: https://github.com/Tremors-Of-War/tremors-of-war/actions/runs/4874645778/jobs/8695847736?pr=47

Reply:
I fixed this by changing the expected type to string[]. That's not necessarily what i wanted to do but I dont know how to get it to recognize the value as Armour/Abilities type